plugins/system_upgrade.py

```diff
@@ -688,12 +701,15 @@ def transaction_download(self):
         self.log_status(_("Download finished."), DOWNLOAD_FINISHED_ID)
 
     def transaction_upgrade(self):
-        Plymouth.message(_("Upgrade complete! Cleaning up and rebooting..."))
-        self.log_status(_("Upgrade complete! Cleaning up and rebooting..."),
+        power_op = "powering off" if self.state.poweroff_after else "rebooting"
+
+        Plymouth.message(_("Upgrade complete! Cleaning up and " + power_op + "..."))
+        self.log_status(_("Upgrade complete! Cleaning up and " + power_op + "..."),
```

Comment:
This change makes the string untranslatable. gettext produces `Upgrade complete! Cleaning up and ` and both `powering off` and `rebooting` are not translated. Just make it separate strings: https://wiki.gnome.org/TranslationProject/DevGuidelines/Never%20split%20sentences

Reply:
Sorry, didn't understand the functionality of the strings at the time. Submitted a fix in #471